(Doc+) Allocation Explain Examples: THROTTLED, MAX_RETRY (#111558)
elastic/elasticsearch: docs/reference/cluster/allocation-explain.asciidoc

<5> The decider which led to the `no` decision for the node.
<6> An explanation as to why the decider returned a `no` decision, with a helpful hint pointing to the setting that led to the decision. In this example, a newly created index has <<indices-get-settings,an index setting>> that requires that it only be allocated to a node named `nonexistent_node`, which does not exist, so the index is unable to allocate.

> ====== Max Retries

Reviewer: ```suggestion
====== Maximum number of retries exceeded
```

> The following response contains an allocation explanation for an unassigned
> primary shard that has reached max allocation retry attempts. 

Reviewer: ```suggestion
The following response contains an allocation explanation for an unassigned
primary shard that has reached the maximum number of allocation retry attempts. 
```

    "last_allocation_status": "no",
  },
  "can_allocate": "no",
  "allocate_explanation": "cannot allocate because allocation is not permitted to any of the nodes",
  "node_allocation_decisions" : [
    {
      "node_id" : "3sULLVJrRneSg0EfBB-2Ew",
      "node_name" : "node_t0",
      "transport_address" : "127.0.0.1:9400",
      "roles" : ["data_content", "data_hot"],
      "node_decision" : "no",
      "store" : {
        "matching_size" : "4.2kb",
        "matching_size_in_bytes" : 4325
      },
      "deciders" : [
        {
          "decider": "max_retry",
          "decision" : "NO",
          "explanation": "shard has exceeded the maximum number of retries [5] on failed allocation attempts - manually call [/_cluster/reroute?retry_failed=true] to retry, [unassigned_info[[reason=ALLOCATION_FAILED], at[2024-07-30T21:04:12.166Z], failed_attempts[5], failed_nodes[[mEKjwwzLT1yJVb8UxT6anw]], delayed=false, details[failed shard on node [mEKjwwzLT1yJVb8UxT6anw]: failed recovery, failure RecoveryFailedException], allocation_status[deciders_no]]]"
        }
      ]
    }
  ]
}
----
// NOTCONSOLE

If decider message indicates a transient allocation issue, use 

> <<cluster-reroute,Cluster Reroute>> to retry allocation. 

Reviewer: ```suggestion
<<cluster-reroute,the cluster reroute API>> to retry allocation. 
```

[source,js]
----
{
  "index" : "my-index-000001",
  "shard" : 0,
  "primary" : false,
  "current_state" : "unassigned",
  "unassigned_info" : {
    "reason" : "NODE_LEFT",
    "at" : "2017-01-04T18:53:59.498Z",
    "details" : "node_left[G92ZwuuaRY-9n8_tc-IzEg]",
    "last_allocation_status" : "no_attempt"
  },
  "can_allocate": "throttled",
  "allocate_explanation": "Elasticsearch is currently busy with other activities. It expects to be able to allocate this shard when those activities finish. Please wait.",
  "node_allocation_decisions" : [
    {
      "node_id" : "3sULLVJrRneSg0EfBB-2Ew",
      "node_name" : "node_t0",
      "transport_address" : "127.0.0.1:9400",
      "roles" : ["data_content", "data_hot"],
      "node_decision" : "no",
      "deciders" : [
        {
          "decider": "throttling",
          "decision": "THROTTLE",
          "explanation": "reached the limit of incoming shard recoveries [2], cluster setting [cluster.routing.allocation.node_concurrent_incoming_recoveries=2] (can also be set via [cluster.routing.allocation.node_concurrent_recoveries])"
        }
      ]
    }
  ]
}
----
// NOTCONSOLE

> This is a transient message which may appear while a large amount of shards are allocating. 

Reviewer: ```suggestion
This is a transient message that might appear when a large amount of shards are allocating. 
```

===== No arguments

If you call the API with no arguments, {es} retrieves an allocation explanation

> for an arbitrary unassigned primary or replica shard, with preference to primary. 

Reviewer: ```suggestion
for an arbitrary unassigned primary or replica shard, returning any unassigned primary shards first. 
```
